lib/reversetunnel/transport.go

```diff
@@ -272,7 +291,15 @@ func (p *transport) start() {
 			}
 
 			p.log.Debugf("Handing off connection to a local %q service.", dreq.ConnType)
-			p.server.HandleConnection(sshutils.NewChConn(p.sconn, p.channel))
+
+			// If dreq has ClientSrcAddr we wrap connection
+			var clientConn net.Conn
+			if dreq.ClientSrcAddr != nil {
+				clientConn = newConnectionWithSrcAddr(sshutils.NewChConn(p.sconn, p.channel), dreq.ClientSrcAddr)
+			} else {
+				clientConn = sshutils.NewChConn(p.sconn, p.channel)
```

Comment:
This is just a function in `api` that returns a type we control; we can just make it hold and return an optional `RemoteAddr` rather than wrap it.

Reply:
Done 👍